.pre-commit-config.yaml

```diff
@@ -123,6 +123,28 @@ repos:
       - id: j2lint
         name: Check for Linting errors on Jinja2 files
 
+  - repo: https://github.com/codespell-project/codespell
+    rev: v2.2.6
+    hooks:
+      - id: codespell
+        name: Check for common misspellings in text files.
+        entry: codespell
+        language: python
+        types: [text]
+        args:
+          - '--ignore-words=ignore-codespell-words.txt'  # Ignore words listed in this file
+        exclude: >  # List of regex patterns for files/directories to exclude
+          (?x)^(
+            .*\.cfg|  # Exclude all .cfg files
+            .*\.svg|  # Exclude all .svg files
+            .*\.excalidraw|  # Exclude all .excalidraw files
+            .*\.json|  # Exclude all .json files
+            .*\.rst|  # Exclude all .rst files
+            python-avd/.*|  # Exclude all files in this directory
+            ansible_collections/arista/avd/docs/release-notes/.*|  # Exclude all files in this directory
+            ansible_collections/arista/avd/changelogs/.*  # Exclude all files in this directory
```

Comment:
Does this mean we have lots of issues in these?

Reply:
Not lot of issues but it will fix the typo of PR titel in changelog file. Let me know if this need to fix
```
ansible_collections/arista/avd/changelogs/changelog.yaml:63: platfom ==> platform
ansible_collections/arista/avd/changelogs/changelog.yaml:129: delimeter ==> delimiter
ansible_collections/arista/avd/changelogs/changelog.yaml:167: seperator ==> separator
ansible_collections/arista/avd/changelogs/changelog.yaml:194: TOI ==> TO, TOY
ansible_collections/arista/avd/changelogs/changelog.yaml:513: plattform ==> platform
ansible_collections/arista/avd/changelogs/changelog.yaml:899: pasword ==> password
ansible_collections/arista/avd/changelogs/changelog.yaml:1238: unter ==> under
ansible_collections/arista/avd/changelogs/changelog.yaml:1248: Incorect ==> Incorrect
```